### ISSUE TITLE
labeled video not generated

### ISSUE THREAD

**Issue by mdandola** [DeepLabCut/DeepLabCut#139]
Hi,
first of all thanks for this awesome tool!

I'm running DeepLabCut 2.0 on linux server, anaconda env.

I trained the network correctly and analyzed a novel video with it (subfolder plot-poses and plot outputs created correctly). The problem occurs when I try to created the labeled video.

This is what I used:
       import deeplabcut
      config_path='mypath'
      deeplabcut.create_labeled_video(config_path,['video_path/video.AVI'],delete=False)

This is the output i get in my command line:
     Starting %  video_path ['/video_path/video.AVI']
     Loading  video_path/video.AVI and data.
     False 0 640 0 480
     1865
     Could not find encoder for codec id 27: Encoder not found
     Duration of video [s]:  9.32 , recorded with  200.0 fps!
     Overall # of frames:  1865 with cropped frame dimensions:  640 480
     Generating frames and creating video.
     100%| 
 ██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1865/1865 [00:31<00:00, 59.77it/s]

Bu the video is not created in the folder "video_path". I looked up the error about the encoder, and I found that ffmpeg needs to be configured with --enable-glp and --enable-libx264. 

1. I tried to do it manually as reported here https://trac.ffmpeg.org/wiki/CompilationGuide/Ubuntu
but nothing changed

2. Then I removed apt installations of ffmpeg, libx264 and libx264-dev and tried to reinstall with conda install -c conda-forge ffmpeg, still nothing changed

3. ffmpeg info of version installed after conda installation:
ffmpeg version 4.1 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 4.8.2 (GCC) 20140120 (Red Hat 4.8.2-15)
  configuration: --prefix=/export/opt/anaconda2/envs/Tracking --disable-doc --disable-openssl --enable-shared --enable-static --extra-cflags='-Wall -g -m64 -pipe -O3 -march=x86-64 -fP        IC' --extra-cxxflags='-Wall -g -m64 -pipe -O3 -march=x86-64 -fPIC' --extra-libs='-lpthread -lm -lz' --enable-zlib --enable-pic --enable-pthreads --enable-gpl --enable-version3 --enabl        e-hardcoded-tables --enable-avresample --enable-libfreetype --enable-gnutls --enable-libx264 --enable-libopenh264
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Hyper fast Audio and Video encoder

thus configuration should be ok.
What am I missing? Any suggestion would be much appreciated, I'm really having troubles to find a solution for this.

thank you!

**Comment by AlexEMG** [comment]
It says: _Could not find encoder for codec id 27: Encoder not found_

So indeed ffmpeg seems to be the problem. This is difficult to troubleshoot from afar. 

You can try to change the coded: and see what you have available.
create_labeled_video(config,videos,shuffle=1,trainingsetindex=0,videotype='avi',save_frames=False,delete=False,displayedbodyparts='all',codec='mp4v'):
by def. it uses: codec='mp4v', which you don't seem to have... 
https://github.com/AlexEMG/DeepLabCut/blob/280e9af7d4e2b676cad496d743304d788bd7d861/deeplabcut/utils/make_labeled_video.py

Perhaps also try to do "save_frames=True" and see if that works (will use a different way to make the video via ffmpeg rather than openCV).

**Comment by MMathisLab** [comment]
Is your env in Anaconda2? Seems the path points to Anaconda2, so... I’m guessing you also have anaconda3 (as deeplabcut wont work in python 2- it needs to be python 3.6) also installed and the default path is not seeing it... you can try installing it inside the already created environment, specifically for python3 by: ‘pip3 install ffmpeg-python’ another work around to double check is to convert your avi to an mp4 (although both avi and mp4 are supported, maybe the avi does indeed have a strange codec, and I’m note sure you specified your video is an avi?). Also, just for future reference help questions we prefer go here: (and given this is not fully specific to dlc, you might get more feedback there anyhow, if you want to post now)
https://forum.image.sc/tags/deeplabcut

**Comment by mdandola** [comment]
Thanks for your quick responses!
- I changed the codec, and set save_frames=True. The error disappeared, frames were saved, but the labelled video generated was empty. No errors showed 
- ffmpeg and libx264 are set correctly in the env, together with anaconda3 and python 3.6. 
I will follow the discussion on rhe image.sc forum. 

thanks again

**Comment by MMathisLab** [comment]
Are the individual frames created (ie labeled correctly)? Also try opening the video in something like VLC if it otherwise looks black, maybe your video player doesn’t have the codec).

**Comment by mdandola** [comment]
yes,
correctly created and stored in the subfolder temp-namevideo

opening the video with quicktime worked! thanks for all the help!

**Comment by Laknath1996** [comment]
Hi,

I am using the DeepLabCut tool in Centos 7. It worked great until the 'analyzing video' step. When I ran the create_labeled_video function, the code ran without displaying any message in the terminal. And when I checked the video folder, it didn't have the frames or the labeled video. Could not figure out what happened there..

Thanks in advance :)

**Comment by MMathisLab** [comment]
Are you sure analyze_video ran? Do you have .h5 files next to videos? There is no error if it doesn’t find a video, so triple check your video path input, video type, etc

**Comment by mukeshnarendran7** [comment]
**Hi, I encountered a similar situation in where my analyze videos(..) runs but then my annotated videos are not annotated (.h5 files present) with the specified labels and also sometimes I get a video which is dark.
Some of the plot poses outcomes for the trajectory are empty too
 #How can I go about resolving this issue? 
Thanks**



**outcome from running analyze videos:**

Config:
{'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7], [8], [9], [10], [11]],
 'all_joints_names': ['nose',
                      'forehead',
                      'Withers',
                      'TailSet',
                      'R_F_Elbow',
                      'R_F_Paw',
                      'L_F_Elbox',
                      'L_F_Paw',
                      'R_B_Hock',
                      'R_B_Paw',
                      'L_B_Hock',
                      'L_B_Paw'],
 'batch_size': 1,
 'crop_pad': 0,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_cow_trimmedAug26/cow_trimmed_...95shuffle1.mat',
 'dataset_type': 'imgaug',
 'deterministic': False,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.7/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 1.0,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'mean_pixel': [123.68, 116.779, 103.939],
 'mirror': False,
 'net_type': 'resnet_50',
 'num_joints': 12,
 'optimizer': 'sgd',
 'pairwise_huber_loss': True,
 'pairwise_predict': False,
 'partaffinityfield_predict': False,
 'regularize': False,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/My '
                    'Drive/Data/dlc-models/iteration-0/cow_trimmedAug26-trainset95shuffle1/test/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}

Using snapshot-22000 for model /content/drive/My Drive/Data/dlc-models/iteration-0/cow_trimmedAug26-trainset95shuffle1
Analyzing all the videos in the directory...
Starting to analyze %  /content/drive/My Drive/Data/videos/VID_20200920_122118.mp4
/content/drive/My Drive/Data/videos  already exists!
Loading  /content/drive/My Drive/Data/videos/VID_20200920_122118.mp4
Duration of video [s]:  7.45 , recorded with  29.82 fps!
Overall # of frames:  222  found with (before cropping) frame dimensions:  1920 1080
Starting to extract posture

230it [00:45,  5.05it/s]

Saving results in /content/drive/My Drive/Data/videos...
Starting to analyze %  /content/drive/My Drive/Data/videos/VID_20200920_115046_limping.mp4
/content/drive/My Drive/Data/videos  already exists!
Loading  /content/drive/My Drive/Data/videos/VID_20200920_115046_limping.mp4
Duration of video [s]:  9.19 , recorded with  29.6 fps!
Overall # of frames:  272  found with (before cropping) frame dimensions:  1920 1080
Starting to extract posture

280it [00:45,  6.19it/s]

Saving results in /content/drive/My Drive/Data/videos...
Starting to analyze %  /content/drive/My Drive/Data/videos/VID_20200920_113834.mp4
/content/drive/My Drive/Data/videos  already exists!
Loading  /content/drive/My Drive/Data/videos/VID_20200920_113834.mp4
Duration of video [s]:  5.53 , recorded with  29.31 fps!
Overall # of frames:  162  found with (before cropping) frame dimensions:  1920 1080
Starting to extract posture

170it [00:28,  5.98it/s]

Saving results in /content/drive/My Drive/Data/videos...
Starting to analyze %  /content/drive/My Drive/Data/videos/VID_20200920_120638_limping.mp4
/content/drive/My Drive/Data/videos  already exists!
Loading  /content/drive/My Drive/Data/videos/VID_20200920_120638_limping.mp4
Duration of video [s]:  10.13 , recorded with  29.62 fps!
Overall # of frames:  300  found with (before cropping) frame dimensions:  1920 1080
Starting to extract posture

310it [00:50,  6.12it/s]

Saving results in /content/drive/My Drive/Data/videos...
The videos are analyzed. Now your research can truly start! 
 You can create labeled videos with 'create_labeled_video'
If the tracking is not satisfactory for some videos, consider expanding the training set. You can use the function 'extract_outlier_frames' to extract a few representative outlier frames.


DLC_resnet50_cow_trimmedAug26shuffle1_22000
